proposals/extensions_v2.md

```diff
@@ -0,0 +1,375 @@
+List<Extension> e3;             // Not legal.  Extension is not a type.
+var v1 = (Extension)receiver;   // Not legal.  Can't can't have a value of extension type.
+```
+
+This is exactly the same as the restrictions on static-types *except* with the carve out that you can use the extension in a cast-syntax or new-expression *only* for lookup purposes, or where a static-class could be used, and nothing else.  Usage in places like `nameof(Extension)` or `typeof(Extension)` would still be fine, as those are places where a static type is allowed.
+
+Note 2. If cast syntax is not desirable here (especially if confuses the idea if extensions are types), we can come up with a new syntactic form.  We are not beholden to the above syntax.
+
+# Future expansion
+
+The above initial strawman solves several major goals for we want for the extensions space:
+
+1. Supporting a much broader set of extension member types.
+2. Having a clean syntax for extension members that matches the regular syntax form (in other words, an extension proeprty still looks like a property).
+3. Ensuring teams can move safely to modern extensions *especially* in environments where source *and* binary compatibility is non-negotiable.
+
+However, there are parts of its core design that are not ideal in the long term which we would like to ensure we can expand on.  These expansions could be released with extensions if time and other resources permit.  Or they could come later and cleanly sit on top of the feature to improve the experience.
+
+These areas are:
+
+## Expansion 1: Syntactic clumsiness and repetition
+
+The initial extension form considers source and binary compatibility as core requirements that must be present to ensure easy migration, allowing codebases to avoid both:
+1. bifurcation; where some codebases adopt modern extensions and some do not.
+2. internal inconsistency; where some codebases must keep around old extensions and new extensions, with confusion about the semantics of how each interacts with the other.
+
+Because classic extension methods have very few restrictions, modern extension methods need to be flexible enough to support all the scenarios which they support.
+
+However, many codebases do not need all the flexibility that classic extension methods afforded.  For example, classic extension methods allow disparate extension methods in a single static class to target multiple different types.  For use cases where that isn't required, we forsee a natural extension (pun intended) where one can translate a modern extension like so:
+
+```c#
+extension E
+{
+    // All extension members extend the same thing:
+
+    public void M() for SomeType { ... }
+    public int P for SomeType { get { ... } }
+    public static operator+(...) for SomeType { ... }
+    // etc
+}
+
+// Can be translated to:
+
+extension E for SomeType
+{
+    public void M() { ... }
+    public int P { get { ... } }
+    public static operator+(...) { ... }
+}
+```
```

Comment:
```suggestion
```c#
extension E
{
    // All extension members extend the same thing:

    public void M() for SomeType str { ... }
    public int P for SomeType str { get { ... } }
    public static operator+(...) for SomeType str { ... }
    // etc
}

// Can be translated to:

extension E for SomeType str
{
    public void M() { ... }
    public int P { get { ... } }
    public static operator+(...) { ... }
}
```
```